--- a/linux-5.18.14/tools/perf/util/scripting-engines/trace-event-python.c	2022-07-23 03:57:00.000000000 -0700
+++ b/linux-5.18.14/tools/perf/util/scripting-engines/trace-event-python.c	2022-07-27 13:37:54.245267843 -0700
@@ -799,8 +799,17 @@
 			_PyLong_FromLong(sample->tid));
 	pydict_set_item_string_decref(dict_sample, "cpu",
 			_PyLong_FromLong(sample->cpu));
-	pydict_set_item_string_decref(dict_sample, "ip",
-			PyLong_FromUnsignedLongLong(sample->ip));
+//	pydict_set_item_string_decref(dict_sample, "ip",
+//			PyLong_FromUnsignedLongLong(sample->ip));
+//OZGURS
+        if (al->map) {
+            pydict_set_item_string_decref(dict_sample, "ip",
+                        PyLong_FromUnsignedLongLong(sample->ip - al->map->start ));//+ al->map->pgoff));
+         } else {
+            pydict_set_item_string_decref(dict_sample, "ip",
+                        PyLong_FromUnsignedLongLong(sample->ip));
+        }
+//OZGURE
 	pydict_set_item_string_decref(dict_sample, "time",
 			PyLong_FromUnsignedLongLong(sample->time));
 	pydict_set_item_string_decref(dict_sample, "period",
